torch_xla/distributed/xla_dist.py

```diff
@@ -297,34 +329,57 @@ def _run_script(script_path, client_worker):
       thread.start()
       threads.append(thread)
 
-    try:
-      for thread in threads:
-        thread.join()
-    except KeyboardInterrupt:
-      pass
-    finally:
-      self._cleanup(script_map)
-
     for thread in threads:
       thread.join()
 
-  def run(self, cmd):
+  def _run_cmd(self, cmd, script_map):
     self.logger.info(
         'Command to distribute: {}'.format(concat_cmd_list(cmd)),
         extra={
             'clientip': '',
             'ordinal': ''
         })
     self.logger.info(
-        'Cluster configuration: {}'.format(self._cluster),
+        f'Cluster configuration: {self._cluster}',
         extra={
             'clientip': '',
             'ordinal': ''
         })
-    script_map = self._prepare_scripts(cmd)
-    self._scp_scripts(script_map)
-    self._start_run(script_map)
+    try:
+      self._scp_scripts(script_map)
+      self._start_run(script_map)
+    except KeyboardInterrupt:
+      pass  # parent process already handles this
```

Comment:
SIGINT does not cross process boundaries.
You should just drop the try/except here.

Reply:
Actually, at first I read that was the case so I didn't have the `try/except` case, but without it I got the following logs:
```
^CProcess Process-1:
2020-01-30 04:02:42  [] Cleaning up processes (takes a couple of seconds)
Traceback (most recent call last):
  File "/anaconda3/envs/torch-xla-nightly/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/anaconda3/envs/torch-xla-nightly/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/torch_xla/distributed/xla_dist.py", line 349, in _run_cmd
    self._start_run(script_map)
  File "/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/torch_xla/distributed/xla_dist.py", line 333, in _start_run
    thread.join()
  File "/anaconda3/envs/torch-xla-nightly/lib/python3.6/threading.py", line 1056, in join
    self._wait_for_tstate_lock()
  File "/anaconda3/envs/torch-xla-nightly/lib/python3.6/threading.py", line 1072, in _wait_for_tstate_lock
    elif lock.acquire(block, timeout):
KeyboardInterrupt
```

Where `Process-1` would be the process running `_run_cmd`.

Reply:
That means that at the time you hit CTRL^C that process owns the TTY.
You can leave the try/except, but print (on stderr) a process message, and then exit with some error code.